File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int arr[n];
    fill_n(arr, n, 1);
    
    int lost_len = lost.size();
    int res_len = reserve.size();
    for (int i = 0; i < lost_len; i++) {
        arr[lost[i] - 1] -= 1;
    }
    for (int j = 0; j < res_len; j++) {
        arr[reserve[j] - 1] += 1;
    }

    for (int k = 0; k < n; k++) {
        if (arr[k] == 0) {
            if (k - 1 >= 0 && arr[k - 1] > 1) {
                arr[k] += 1;
                arr[k - 1] -= 1;
                answer += 1;
            } else if (k + 1 < n && arr[k + 1] > 1) {
                arr[k] += 1;
                arr[k + 1] -= 1;
                answer += 1;
            }
        } else {
            answer += 1;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    clothes = [1 for _ in range(n)]
    
    for idx in reserve:
        clothes[idx - 1] += 1
    for idx in lost:
        clothes[idx - 1] -= 1
    
    for idx, val in enumerate(clothes):
        if val == 0:
            if 0 <= idx - 1:
                if clothes[idx - 1] > 1:
                    clothes[idx - 1] -= 1
                    clothes[idx] += 1
                    continue
            if idx + 1 < n:
                if clothes[idx + 1] > 1:
                    clothes[idx + 1] -= 1
                    clothes[idx] += 1
    
    return n - clothes.count(0)